tests/MagicMethodAnnotationTest.php

```diff
@@ -817,6 +960,7 @@ class A {}
                     class B extends A {}
                     function consumeInt(int $i): void {}
 
+                    /** @psalm-suppress UndefinedMethod */
                     consumeInt(B::bar());',
```

Comment:
I'd suggest adding another test:

```php
[
   'code' => <<<'PHP'
   <?php
   /** @method static int bar() */
   class A {}
   class B extends A {}

   /** @psalm-suppress UndefinedMethod */
   $a = B::bar(123, "whatever");
   PHP,
   'assertions' => [
       '$a===' => 'mixed',
   ],
],
```
and one more:
```php
[
   'code' => <<<'PHP'
   <?php
   /** @method static int bar() */
   class A {}
   class B extends A {}

   /** @psalm-suppress UndefinedMethod */
   $a = B::bar(function_does_not_exist(123));
   PHP,
   'error_message' => 'UndefinedFunction',
],
```

This will make sure that ignoring the method also ignores its parameters / return value, but the statements inside the argument list are still validated (thus `UndefinedFunction` in the second case)

Reply:
I agree, the test has become confusing.

It's probably better to fix this test. I think your version with `assertions` is better.
But I would remove the confusing code with `@psalm-suppress UndefinedMethod`, and replace it with `error_levels`.

```
[
    'code' => <<<'PHP'
        <?php
        /**
         * @method static int bar()
         */
        class A {}
        class B extends A {}

        $a = B::bar();
        PHP,
    'assertions' => [
        '$a===' => 'int',
    ],
    'error_levels' => [
        'UndefinedMethod',
    ],
],
```

@weirdan , what do you think about this version of the code?